### PR TITLE
Implement double backward of SLSTM function

### PR DESCRIPTION
**Pull request by tohmae**
I implement double backward of SLSTM function.
SLSTM still can not support higher-order derivatives,
but it supports double backward.
#4449

### CODE REVIEW

**Comment by tohmae** [comment]
I calculated the second order derivative of SLSTM function by myself.
So was this derivative wrong ?
If you want to delete this PR, I delete this PR.

File: chainer/functions/activation/slstm.py [hunk 1]
```diff
@@ -176,12 +205,126 @@ def backward(self, inputs, grad_outputs):
...
+            = slstm_grad_grad(c_prev1, a1, i1, f1,
+                              c_prev2, a2, i2, f2, o, c, gc, gh,
+                              ggc_prev1, gga1, ggi1, ggf1, ggo1,
+                              ggc_prev2, gga2, ggi2, ggf2, ggo2)
+        return gc_prev1, gc_prev2, gx1, gx2, gc_next, ggc, ggh
+
+
+def _cupy_sigmoid(x):
+    half = x.dtype.type(0.5)
+    return cuda.fusion.tanh(x * half) * half + half
```
Comment (toslunar) [inline chainer/functions/activation/slstm.py]: GPU tests failed on this line: `AttributeError: 'module' object has no attribute 'fusion'`

**Comment by tohmae** [comment]
Thank you for your comment.
In master cuda_fusion.py was removed recently (#5144).
So I use cuda.get_array_module in fused function like lstm function.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**